src/pointer-analysis/value_set_dereference.cpp

```diff
@@ -202,7 +202,12 @@ bool value_set_dereferencet::dereference_type_compare(
   const typet &dereference_type) const
 {
   if(dereference_type.id()==ID_empty)
-    return true; // always ok
+    return true; // always ok (anything can be dereferenced to void type)
+
+  if(
+    object_type.id() == ID_pointer && dereference_type.id() == ID_pointer &&
```

Comment:
The first if handles `void*`, this one handles `void**` - what about `void***`? (not possible I assume in Java but perhaps in C?)

Reply:
Good points. I took a shot at it in the next commit, let me know what you think.